app/javascript/chat/components/ConnectStateProvider.jsx

```diff
@@ -0,0 +1,22 @@
+import { h, createContext } from 'preact';
```

Comment:
@narender2031, @citizen428, since we're going to go with useReducer and context, I'll merge my PR, #9799, first.

Reply:
sure